### DISCUSSION TITLE
Connexion de LibreChat à Perplexity.

### DISCUSSION THREAD

**Discussion by Angersrezo** [danny-avila/LibreChat#1894] [Help Wanted]
Has anyone connected LibreChat to Perplexity.
I'm looking for librechat.yaml files that might have worked.

**Comment by Angersrezo** [comment]
J’ai tenté de connecter LibreChat à Perplexity.
Voici mon librechat.yaml



version: 1.0.3

cache: true
endpoints:
  custom:
    - name: "Perplexity"
      apiKey: "pplx-db58555ad091f9ca7849ae60c780486XXXXXXXXXX"
      baseURL: "https://api.perplexity.ai"
      models:
        default: ["sonar-small-chat"]
        fetch: false
        titleConvo: true
        titleModel: "sonar-small-chat"
        summarize: false
        forcePrompt: false
        modelDisplayLabel: "Perplexity - sonar-small-chat"
        addParams:
          # Ajoutez ici des paramètres supplémentaires spécifiques à Perplexity si nécessaire
        dropParams: ["stop", "user", "frequency_penalty", "presence_penalty"]
        ## dropParams:
        ##  - "frequency_penalty"
        ##  - "presence_penalty"



This is the error message I get back.
Has anyone experienced this before?

![image](https://github.com/danny-avila/LibreChat/assets/160249919/2a83657d-6523-4cad-a64f-324666a3bb25)
![image](https://github.com/danny-avila/LibreChat/assets/160249919/e51ea032-fa86-436c-9b6c-f2e780f170aa)

**Reply by Nahaboo** [reply]
found it. Update the models : https://docs.perplexity.ai/guides/model-cards
    - name: "Perplexity"
      apiKey: "pplx-dddxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxca"
      baseURL: "https://api.perplexity.ai/"
      models:
        default: [
          "llama-3.1-sonar-small-128k-online",
          "llama-3.1-sonar-large-128k-online",
          "llama-3.1-sonar-huge-128k-online"
          ]
        fetch: false # fetching list of models is not supported
      titleConvo: true
      titleModel: "llama-3-sonar-small-32k-chat"
      summarize: false
      summaryModel: "llama-3-sonar-small-32k-chat"
      forcePrompt: false
      dropParams: ["stop", "frequency_penalty"]
      modelDisplayLabel: "Perplexity"

**Comment by Nahaboo** [comment]
I have the same issue, did you find a solution ?

**Comment by Angersrezo** [comment]
I've been doing this for a while.
I remember having a lot of problems with something that looked like invisible characters left behind when retyping by hand, and also going through OpenRouter.

**Reply by Nahaboo** [reply]
ah... then i don't know because by changing the models i could solve the issue.
